### PR TITLE
Further utils edits

### PR DESCRIPTION
**Pull request by cdonnay**
- preserve the resolutions of tiebreaks
- ensure that ballots are of appropriate type: score v rank
- add function for computing score dictionary from score ballot
- add/edit tests

### CODE REVIEW

File: src/votekit/utils.py [hunk 1]
```diff
@@ -339,7 +388,9 @@ def tiebroken_ranking(
     ranking: tuple[frozenset[str], ...],
     profile: Optional[PreferenceProfile] = None,
     tiebreak: str = "random",
-) -> tuple[frozenset[str], ...]:
+) -> tuple[
```
Comment (peterrrock2) [inline src/votekit/utils.py:391]: You have a `random_seed` parameter in `tiebreak_set`. Shouldn't that parameter be here as well if you intend to pass it? Of course, you can get rid of the parameter entirely by telling the user that to make the election deterministic, they need to set the PYTHONHASHSEED environment variable as well as the numpy seed before doing any of their computation.

File: src/votekit/utils.py [hunk 2]
```diff
@@ -351,23 +402,27 @@ def tiebroken_ranking(
...
     new_ranking: list[frozenset[str]] = [frozenset()] * len(
         [c for s in ranking for c in s]
     )
 
     i = 0
+    tied_dict = {}
     for s in ranking:
         if len(s) > 1:
             tiebroken = list(tiebreak_set(s, profile, tiebreak))
+            tied_dict[s] = tuple(tiebroken)
```
Comment (peterrrock2) [inline src/votekit/utils.py:419]: Small nit-pick. Rather than converting to a list and then to a tuple, it would probably be best to use the returned tuple directly (I believe that `tiebreak_set` returns a tuple)

```
tied_dict[s] = tiebreak_set(s, profile, tiebreak)
tiebroken = list(tied_dict[s])
```

File: tests/test_utils.py
Comment (peterrrock2) [inline tests/test_utils.py:1]: Missing tests for instances of the TypeError "Ballots must have rankings"

Comment (peterrrock2) [reply tests/test_utils.py:1]: Missing test for `ValueError(f"Ballot {ballot} has an empty ranking position.")` in `score_profile_from_rankings`

**Review by peterrrock2 — CHANGES_REQUESTED** [review]
Looking good! Just a couple of small notes for this one.

File: src/votekit/utils.py [hunk 3]
```diff
@@ -85,29 +88,55 @@ def remove_cand(
...
             scrubbed_ballots[i] = Ballot(
                 ranking=tuple(new_ranking), weight=ballot.weight
             )
 
-    # easiest way to condense ballots
-    clean_profile = PreferenceProfile(
-        ballots=tuple([b for b in scrubbed_ballots if isinstance(b, Ballot)])
-    ).condense_ballots()
+        elif len(new_scores) > 0:
+            scrubbed_ballots[i] = Ballot(weight=ballot.weight, scores=new_scores)
```
[the remark this answers is not in the document]
Comment (cdonnay) [reply src/votekit/utils.py:116]: Else the ballot is exhausted! Added that else statement.

File: src/votekit/utils.py [hunk 4]
```diff
@@ -285,13 +331,14 @@ def borda_scores(
...
 
-    return score_profile(profile, score_vector, to_float)
+    return score_profile_from_rankings(profile, score_vector, to_float)
 
 
 def tiebreak_set(
     r_set: frozenset[str],
     profile: Optional[PreferenceProfile] = None,
     tiebreak: str = "random",
+    random_seed: Optional[int] = None,
```
[the remark this answers is not in the document]
Comment (cdonnay) [reply src/votekit/utils.py:341]: Whoops!

**Review by peterrrock2 — APPROVED** [review]
